src/.clang-tidy

```diff
@@ -0,0 +1,28 @@
+---
+Checks:          '-*,
+                 # modernize-use-auto,
```

Comment:
I know we have auto in https://github.com/nodejs/node/blob/master/CPP_STYLE_GUIDE.md#using-auto. But it's more like a principle. Should we apply this rule and talk the options https://clang.llvm.org/extra/clang-tidy/checks/modernize-use-auto.html#cmdoption-arg-mintypenamelength.

cc @addaleax @joyeecheung @refack

Reply:
I am -1 to apply modernize-use-auto for our code base. I am fine with patches that use auto, but I don't think switching into auto from explicit types improve the readability especially during code reviews.

Reply:
Let's not enable this rule then :)